src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/online-unit-form/online-unit-form.component.ts

```diff
@@ -76,27 +78,9 @@ export class OnlineUnitFormComponent implements OnInit, OnChanges {
     }
 
     ngOnChanges(): void {
-        this.initializeForm();
-        if (this.isEditMode && this.formData) {
-            this.setFormValues(this.formData);
-        }
-    }
-
-    ngOnInit(): void {
-        this.initializeForm();
-    }
-
-    private initializeForm() {
-        if (this.form) {
-            return;
+        if (this.isEditMode() && this.formData()) {
+            this.setFormValues(this.formData()!);
```

Comment:
You use an effect for this and get rid of the ngOnChanges()
```suggestion

        constructor() {
          effect(() => {
            const formData = this.formData();
            // check in setFormValues if edit mode is enabled
            untracked(() => this.setFormValues(this.formData));
          });
        }
        if (this.isEditMode() && this.formData()) {
            this.setFormValues(this.formData()!);
```

Reply:
I am not sure about the benefit of the suggested change.
I do think that a ngOnInit call is much easier to understand and less error-prone. Can you elaborate the advantages of the suggested approach?

Is the suggested approach actually semantically equal, even if setFormValues is untracked, wouldn't the effect still be triggered whenever formData is changed?
I assume in the suggestion lines 89 + 90 shall rather be deleted? Because this will raise errors for my understanding.
This would then also mean ignoring whether we are in editMode or not?

Reply:
Discussed with @JohannesWt, we will have a offline chat about it and keep it as it is for now within this PR